apps/code-examples/src/app/code-examples/ag-grid/data-grid/paging/data-grid-paging-demo.component.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { ActivatedRoute, NavigationEnd, Router } from '@angular/router';
+import { SkyAgGridService, SkyCellType } from '@skyux/ag-grid';
+
+import {
+  ColDef,
+  GridApi,
+  GridOptions,
+  GridReadyEvent,
+  ValueFormatterParams,
+} from 'ag-grid-community';
+import { Subscription } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+
+import { SKY_AG_GRID_DEMO_DATA } from './data-grid-paging-demo-data';
+
+@Component({
+  selector: 'app-data-grid-paging-demo',
+  templateUrl: './data-grid-paging-demo.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DataGridPagingDemoComponent implements OnInit, OnDestroy {
+  public currentPage = 1;
+
+  public readonly pageSize = 3;
+  public columnDefs: ColDef[] = [
+    {
+      field: 'name',
+      headerName: 'Name',
+    },
+    {
+      field: 'age',
+      headerName: 'Age',
+      type: SkyCellType.Number,
+      maxWidth: 60,
+    },
+    {
+      field: 'startDate',
+      headerName: 'Start date',
+      type: SkyCellType.Date,
+      sort: 'asc',
+    },
+    {
+      field: 'endDate',
+      headerName: 'End date',
+      type: SkyCellType.Date,
+      valueFormatter: this.endDateFormatter,
+    },
+    {
+      field: 'department',
+      headerName: 'Department',
+      type: SkyCellType.Autocomplete,
+    },
+    {
+      field: 'jobTitle',
+      headerName: 'Title',
+      type: SkyCellType.Autocomplete,
+    },
+  ];
+
+  public gridApi: GridApi | undefined;
+  public gridData = SKY_AG_GRID_DEMO_DATA;
+  public gridOptions: GridOptions;
+  public searchText = '';
+
+  #subscriptions = new Subscription();
+
+  constructor(
+    private agGridService: SkyAgGridService,
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private changeDetector: ChangeDetectorRef
```

Comment:
We could use `#agGridSvc = inject(SkyAgGridService)` etc. for these.

Reply:
Good call. Switched.